spec/request/api/v1/questions/create_question_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+describe 'POST /api/v1/questions', type: :request do
+  let(:user) { create(:user) }
+  let(:question) { build(:question, user: user) }
+  let(:params) do
+    {
+      question: {
+        email_from: question.email_from,
+        message: question.message
+      }
+    }
+  end
+
+  context 'when the user is not logged in' do
+    it 'returns unauthorized response' do
+      post api_v1_questions_path, params: params, as: :json
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
+
+  context 'when request is valid' do
+    subject do
+      post api_v1_questions_path, params: params, headers: user.create_new_auth_token, as: :json
+    end
+
+    it 'returns a successful response' do
+      subject
+      expect(response).to be_successful
+    end
+
+    it 'creates a question with correct attributes' do
+      subject
+      expect(Question.last).to have_attributes(email_from: question.email_from, message: question.message)
+    end
+
+    it 'sends an email' do
+      expect { subject }.to have_enqueued_job.on_queue('mailers')
```

Comment:
To test that the email was sent, I think it's better to use:
```
expect { subject }.to change { ActionMailer::Base.deliveries.count }.by(1)
```

Reply:
I tried that. Did not work. I guess because it does not send the email really, only add it to a queue to send it later.

Reply:
Maybe you are missing `config.action_mailer.delivery_method = :test` on the test environment file

Reply:
I wish i had miss that line, but It's right there, in the test.rb file